mlir/lib/Dialect/Bufferization/IR/BufferizableOpInterface.cpp

```diff
@@ -600,8 +600,12 @@ bool AnalysisState::canOmitTensorCopy(OpOperand &opOperand) const {
 }
 
 bool AnalysisState::isInPlace(OpOperand &opOperand) const {
-  // ToMemrefOps are always in-place.
-  if (isa<ToMemrefOp>(opOperand.getOwner()))
+  // We should always inplace operands that must be inplaced.
+  auto bufferizableOp =
+      getOptions().dynCastBufferizableOp(opOperand.getOwner());
+  if ((bufferizableOp &&
+       bufferizableOp.mustBufferizeInPlace(opOperand, *this)) ||
```

Comment:
This is problematic because `AnalysisState` won't be able to detect cases where multiple ops, that are all marked as `mustBufferizeInPlace`, cause a RaW conflict.

When running with `OneShotBufferizationState`, we are able to detect such cases:
```
not bufferizable under the given constraints: cannot avoid RaW conflict
```

What is your use case? There are no atomic operations that operate on tensors as far as I know.